src/asio.cpp

```diff
@@ -106,11 +106,10 @@ namespace fc {
                {
                  fc::thread::current().set_name("asio");
                  
-                 BOOST_SCOPE_EXIT(void)
+                 //BOOST_SCOPE_EXIT(void)
```

Comment:
Why?

@jmjatlanta this change looks like something we discussed a few days ago. Please check.

Reply:
We discussed this same section of code, but it appears we were looking at solving a different problem. I believe as you do, that this is a misplaced PR, or at a minimum, needs more explanation of what issues are attempting to be solved.